### PR TITLE
Adjust core implementation

### PR DESCRIPTION
**Pull request by alperen-commencis**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new analytics integration using the Adjust SDK for event tracking.
  - Added a login success event for better tracking through Adjust.

- **Enhancements**
  - Updated Firebase Messaging to a new static initialization approach for improved performance and maintainability.
  - `NeoCore` initialization now supports an Adjust app token for seamless integration with the Adjust SDK.

- **Dependencies**
  - Updated dependencies to replace `posthog_flutter` with `adjust_sdk` for analytics.

### CODE REVIEW

File: lib/core/analytics/neo_adjust/events/neo_adjust_events.dart [hunk 1]
```diff
@@ -0,0 +1,7 @@
+enum NeoAdjustEvent {
+  // STOPSHIP: Update event id from Adjust panel when event created
+  neoLoginSuccess("testId");
```
Comment (yildirayyorulmaz) [inline lib/core/analytics/neo_adjust/events/neo_adjust_events.dart:3]: suggestion: loginSuccess yeterli bence tum enumlara 'neo' prefixi eklemeyelim

Comment (alperen-commencis) [reply lib/core/analytics/neo_adjust/events/neo_adjust_events.dart]: applied

File: lib/core/analytics/neo_adjust/events/neo_adjust_events.dart [hunk 2]
```diff
@@ -0,0 +1,7 @@
+enum NeoAdjustEvent {
+  // STOPSHIP: Update event id from Adjust panel when event created
+  loginSuccess("testId");
```
Comment (reyhanlioglu) [inline lib/core/analytics/neo_adjust/events/neo_adjust_events.dart:3]: burası test olarak kalmış

File: lib/core/analytics/neo_adjust/neo_adjust.dart [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  Future<void> init({required String appToken}) async {
+    final String? deviceId = await DeviceUtil().getDeviceId();
+
+    final AdjustConfig adjustConfig =
+        AdjustConfig(appToken, kDebugMode ? AdjustEnvironment.sandbox : AdjustEnvironment.production)
+          ..externalDeviceId = deviceId;
+
+    Adjust.start(adjustConfig);
+  }
```
[the remark this answers is not in the document]
Comment (reyhanlioglu) [reply lib/core/analytics/neo_adjust/neo_adjust.dart:21]: bunu da yapabiliriz

**Review by reyhanlioglu — APPROVED** [review]

**Pull request merged**